sysutils/joshuto/Portfile

```diff
@@ -0,0 +1,133 @@
+    structopt                       0.3.21  5277acd7ee46e63e5168a80734c9f6ee81b1367a7d8772a2d765df2a3705d28c \
+    structopt-derive                0.4.14  5ba9cdfda491b814720b6b06e0cac513d922fc407582032e8706e9f137976f90 \
+    syn                             1.0.72  a1e8cdbefb79a9a5a65e0db8b47b723ee907b7c7f8496c76a1770b5c310bab82 \
+    termion                          1.5.6  077185e2eac69c3f8379a4298e1e07cd36beb962290d4a51199acf0fdc10607e \
+    textwrap                        0.11.0  d326610f408c7a4eb6f51c37c330e496b08506c9457c9d34287ecc38809fb060 \
+    time                            0.1.43  ca8a50ef2360fbd1eeb0ecd46795a87a19024eb4b53c5dc916ca1fd95fe62438 \
+    toml                             0.5.8  a31142970826733df8241ef35dc040ef98c679ab14d7c3e54d827099b3acecaa \
+    trash                            1.3.0  90df96afb154814e214f37eac04920c66886fd95962f22febb4d537b0dacd512 \
+    tui                             0.15.0  861d8f3ad314ede6219bcb2ab844054b1de279ee37a9bc38e3d606f9d3fb2a71 \
+    unicode-segmentation             1.7.1  bb0d2e7be6ae3a5fa87eed5fb451aff96f2573d2694942e40543ae0bbe19c796 \
+    unicode-width                    0.1.8  9337591893a19b88d8d87f2cec1e73fad5cdfd10e5a6f349f498ad6ea2ffb1e3 \
+    unicode-xid                      0.2.2  8ccb82d61f80a663efe1f787a51b16b5a51e3314d6ac365b08639f52387b33f3 \
+    users                           0.11.0  24cc0f6d6f267b73e5a2cadf007ba8f9bc39c6a6f9666f8cf25ea809a153b032 \
+    utf8parse                        0.1.1  8772a4ccbb4e89959023bc5b7cb8623a795caa7092d99f3aa9501b9484d4557d \
+    vec_map                          0.8.2  f1bddf1187be692e79c5ffeab891132dfb0f236ed36a43c7ed39f1165ee20191 \
+    version_check                    0.9.3  5fecdca9a5291cc2b8dcf7dc02453fee791a280f3743cb0905f8822ae463b3fe \
+    void                             1.0.2  6a02e4885ed3bc0f2de90ea6dd45ebcbb66dacffe03547fadbb0eeae2770887d \
+    wasi                          0.9.0+wasi-snapshot-preview1  cccddf32554fecc6acb585f82a32a72e28b48f8c4c1883ddfeeeaa96f7d8e519 \
+    wasi                          0.10.2+wasi-snapshot-preview1  fd6fbd9a79829dd1ad0cc20627bf1ed606756a7f77edff7b66b7064f9cb327c6 \
+    which                            4.1.0  b55551e42cbdf2ce2bedd2203d0cc08dba002c27510f86dab6d0ce304cba3dfe \
+    whoami                           0.9.0  7884773ab69074615cb8f8425d0e53f11710786158704fca70f53e71b0e05504 \
+    winapi                           0.3.9  5c839a674fcd7a98952e593242ea400abe93992746761e38641405d28b00f419 \
+    winapi-i686-pc-windows-gnu       0.4.0  ac3b87c63620426dd9b991e5ce0329eff545bccbbb34f3be09ff6fb6ab51b7b6 \
+    winapi-x86_64-pc-windows-gnu     0.4.0  712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f \
+    xdg                              2.2.0  d089681aa106a86fade1b0128fb5daf07d5867a509ab036d99988dec80429a57
+
+build.pre_args      --release -v -j${build.jobs}
```

Comment:
Hey @judaew - you provide `cargo.crates` above, but at the same time you set `build.pre_args` without the `--frozen` flag... 

`--frozen` is required to make `cargo` not attempt to download deps from the Internet and only use what's provided by `cargo.crates`.  Is there any reason why you felt you needed to override `build.pre_args`?

Reply:
Hmm, if I use `--frozen` (default PG settings), then the port is not built. Does this mean it cannot be built without Internet access or am I doing something wrong?

```
:debug:build system:  cd "/opt/local/var/macports/build/.../joshuto/work/joshuto-0.9.0" && /opt/local/bin/cargo build --release --frozen -v -j8 
:info:build error: the lock file /opt/local/var/macports/build/.../kamiyaa-joshuto-65f7aaf/Cargo.lock needs to be updated but --frozen was passed to prevent this
:info:build If you want to try to generate the lock file without accessing the network, use the --offline flag.
:info:build Command failed:  cd "/opt/local/var/macports/build/.../joshuto/work/joshuto-0.9.0" && /opt/local/bin/cargo build --release --frozen -v -j8 
:info:build Exit code: 101
```

Reply:
This means that the `Cargo.lock` file that you generated your `cargo.crates` from is not matching the original `Cargo.toml`.  As a result of this it wants to connect to the Internet to fetch what it thinks is missing.

You can leave as is, but if you want to have everything work using the default `--frozen` configuration, then you will need to regenerate `Cargo.lock` and then regenerate `cargo.crates` from that.

You would then need to include a patch in the Portfile that patches `Cargo.lock` when the port is extracted.

To regenerate the `Cargo.lock`, you would need to do something like:
1. `sudo port extract joshuto`
2. `cd $(port work joshuto)/joshuto...`
3. `sudo -u macports HOME=../.home cargo regenerate-lockfile`

Now with that new `Cargo.lock` file, you can create the patch file and create a correct `cargo.crates` section.

Reply:
Thanks so much. Earlier with other ports, I also had a similar problem, but now I will know how to solve it.